connector/src/main/java/org/geysermc/connector/network/translators/world/block/BlockStateValues.java

```diff
@@ -335,6 +336,26 @@ public static boolean canPistonDestroyBlock(int state)  {
         return BlockRegistries.JAVA_BLOCKS.getOrDefault(state, BlockMapping.AIR).getPistonBehavior() == PistonBehavior.DESTROY;
     }
 
+    public static boolean canPistonMoveBlock(int javaId, boolean isPushing) {
+        if (javaId == BlockStateValues.JAVA_AIR_ID) {
+            return true;
+        }
+        // Pistons can only be moved if they aren't extended
+        if (PistonBlockEntityTranslator.isBlock(javaId)) {
+            return !BlockStateValues.getPistonValues().get(javaId);
```

Comment:
You can remove the specific declaration to BlockStateValues here since we're in the class file.